### DISCUSSION TITLE
Greener coding: project Q&A

### DISCUSSION THREAD

**Discussion by thibaudcolas** [wagtail/gsoc#93] [General]
I thought I would use this thread to collate all of the questions I’ve had about this project so far, so everyone has access to the same information.

**Comment by thibaudcolas** [comment]
Q: What led to the creation of https://github.com/green-coding-berlin/bakerydemo-gold-benchmark?

**Reply by thibaudcolas** [reply]
A: You can read more about our history of partnership with the Green Web Foundation on https://github.com/wagtail/wagtail/discussions/8843.

They provided benchmarking tools to evaluate Wagtail. We (me) have provided the demo website linked here. It’s completely fine to use the information on there, and suggested improvements, in your proposal. Doing this kind of research is valuable, we don’t expect people to come up with all of this on their own – just appreciate demonstrating some judgement as far as which improvements are doable within a given time frame.

**Comment by thibaudcolas** [comment]
Q: On measurement of improvements – do we generate better scripts for Greenframe to use or are we looking for a different method other than the Greenframe approach?

**Reply by thibaudcolas** [reply]
A: what we’ve done so far is only benchmarking of Wagtail’s current environmental footprint. If we want to improve our benchmarking, we can create better scripts / user other tools. But we’d also like to move on to making improvements in the CMS at this point.

**Comment by thibaudcolas** [comment]
Q: There was a mention of implementing dark themes and possibly different contrast modes even though the heading of the project was "Greener Coding". Wagtail is planning to move forward with it outside GSoC, is this still relevant?

**Reply by thibaudcolas** [reply]
A: Yes, though note this is just one example of the type of work that could happen in this project. We’re moving forward with a dark theme, at least in Wagtail itself. This could still be part of the Greener Coding project if you’re interested in this – measuring the impact of those changes, doing similar changes on some of our other projects (for example [wagtail.org](http://wagtail.org/)), or the bakerydemo site. Or introducing a high-contrast theme with quantified energy efficiency.

**Comment by thibaudcolas** [comment]
Q: I did some research and figured out few ways to make Wagtail "greener", I was realized that caching and handling media content via CDNs and packages like Whitenoise (inspired from wagtail-gold-benchmark) helped in making wagtail efficient, which means lesser carbon impact. Now I was wondering why some of these changes are not already implemented? Will it create some issues?

**Reply by thibaudcolas** [reply]
A: Wrong suggestions are ok as long as you made some effort to validate them! There are two reasons why those types of things might not have happened already:

- Quite a few of those improvements are project-specific. We want Wagtail to be flexible, so for example can’t force everyone using Wagtail to use Whitenoise. Though we could definitely make sure it’s well documented how to do so.
- Planning – Performance is something we only look at occasionally, so projects like Greener Coding are a great occasion to focus on this for a bit.

**Comment by thibaudcolas** [comment]
Q: You had asked to produce a report on wagtail's climate impact. But this is highly variable. We can only check for deployed sites and sites vary from one another. What is Wagtail planning to do on this report? Are we going to establish a standard, like BakeryDemo to compare with?

**Reply by thibaudcolas** [reply]
A: That’s the right questions to ask yourself. We ask for a report for two reasons: to see how people approach this type of task, and get fresh ideas. https://github.com/thibaudcolas/bakerydemo-gold-benchmark is the current standard we’ve established, but I think a report on any live site would be good too. This could be as simple as running EcoPing on a page of a Wagtail site, and reviewing which suggestions might be related to Wagtail.

**Comment by thibaudcolas** [comment]
Q: Is there pre-existing attempts at quantifying the impact of potential improvements to Wagtail?

**Reply by thibaudcolas** [reply]
A: Yes, we did two studies, which we haven’t shared publicly yet:

- A comparison of [possible sustainability improvements](https://docs.google.com/spreadsheets/d/1TiX4D-nFd2F3vlFljZ0PYNLBq3ZPLeeQkDh8QH7LvRw/edit#gid=0) that should be doable on a large amount of Wagtail sites
- A review of [Wagtail Lighthouse performance audit scores vs. average site scores](https://docs.google.com/spreadsheets/d/1juEvqAe2evhBChCgODII1zPTXPpe6wiRInOAvQozjUQ/edit#gid=1256810622). This shows where Wagtail has the most room to improve compared to _average_ websites.

**Comment by vmmuthu31** [comment]
Hi, I would like to contribute for this project.

**Reply by thibaudcolas** [reply]
Hey, cool, do you have any questions about the process?

**Comment by SAMAD101** [comment]
Thank you @thibaudcolas , this thread is an asset for contributors like me.
